### ISSUE TITLE
Freeze on Ubuntu 22.04

### ISSUE THREAD

**Issue by yuqiang-yuan** [FontManager/font-manager#295]
**Describe the bug**
The Font Manager window freezes on Ubuntu 22.04 with no font loaded and shown.

**System Information**
 - OS: Ubuntu 22.04
 - Desktop Environment: GNOME 42.1
 - Font Manager version:  0.8.8
 - Font Manager installation method: apt

**Additional context**
I installed Font Manager on Ubuntu 20.04 and evetything goes ok, then I upgrade to Ubuntu 22.04 via `do-release-upgrade` Font Manager GUI freezes often

![Screenshot from 2022-06-17 01-52-26](https://user-images.githubusercontent.com/15667885/174134833-5d8a75f7-89b9-42e5-ac17-423ac5928f83.png)

**Comment by JerryCasiano** [comment]
Is this still an issue ?

**Comment by yuqiang-yuan** [comment]
@JerryCasiano I've upgraded to Ubuntu 22.04.1 and switch to Wayland and it seems FontManager works fine.

**Comment by mdrmike** [comment]
@JerryCasiano its still an issue for me... I have vanilla Ubuntu 22.04.  Just installed this morning. :-(

While not exactly frozen, the app isn't functional.  I can click `preferences` cog, and it'll show preferences, I can select `manage` | `browse` | `compare`, etc but I just can't change or select anything in main window. Here's an example.    

![Screencast from 12-28-2022 09:21:43 AM-optimized](https://user-images.githubusercontent.com/259940/209854982-76515bd0-b1d2-4324-a778-c6f6fa03c79b.gif)


I thought maybe problem was related to number of fonts on my machine?    

I've added a `~/.fonts` folder in my home directory.  Inside my `.fonts` folder I have a googlewebfonts folder which has a symlink to a local git repo of the official google webfonts. Inside I have links to each folder in `/googlewebfonts/ofl/` _except_ `notocoloremoji`, (which interferes with default `gnome-characters`). 

I've also a few miscellaneous, mostly opensource/free fonts in additional `.fonts/subfolders` that I've acquired over the years.

![Screenshot from 2022-12-28 09-09-52](https://user-images.githubusercontent.com/259940/209847880-41f3dfdc-3920-4a71-81ee-999f4efd731e.png)

**Comment by JerryCasiano** [comment]
First thing to try is deleting your folder of links, obviously.

That doesn't make sense to do anyway you're much better off adding the actual directory as a source in the preferences pane. That's exactly the use case for that feature.

I would also temporarily move all those font folders into a different directory then delete ~/.cache/font-manager and try again.

If you get a normal experience add each font folder back one at a time while restarting the application until you either encounter an issue again or everything works normally and all are back in place. Then add the Google git repo as a source.

If that doesn't help, post back but don't forget to include installation details, OS, DE, etc.

**Comment by mdrmike** [comment]
### Background

Hmm, before I submitted a comment above -- I tried opening & closing the program and letting it run for 15min (or more), in thoughts maybe it was overwhelmed by number of fonts.

None of it made a difference so I [documented my experience (above)](https://github.com/FontManager/font-manager/issues/295#issuecomment-1366835997).


### Resolution Process

- Per suggestion, I removed all my fonts.  Started Font-Manager and it worked!  Yay!
- Next, I reintroduced all of my fonts, except the google fonts repo I have locally.  Again, worked.  Yay!
- Then, I began re-introducing googlewebfonts, EXACTLY as before, but only a limited number of font folders at a time alphabetically. First all font folders that began with A and B.  Started font manager and it worked.  Yay!
- I continued, and it worked for B-C, tested, worked, then E-F, tested, worked. Then I got more confident and added like H-K.  And THEN Font Manager froze!  So I removed the last batch. Restarted and Font-Manager worked. I thought I found the font causing the problem. I began reintroducing each font folder one letter at time.  By the end, they all worked. I was confused.  But continued.


### Bug Found 

Eventually I ran into a similar problem. The key is **adding too many new fonts at once!**  It crashes Font Manager (per my report above).  Adding fewer font folders/files at once resolves the issue.  But definitely a bug hat it can't handle hundreds of fonts at once. I now have the exact setup I wanted, with approximately 1600 additional fonts, mostly Google Web Fonts.   



**Final note:**   
I mentioned in my originally issue that I was symlinking. I suspected symlinking to folders made ZERO difference. I was correct. My original setup is exactly the same as when I initially had a problem with Font Manager. But now it works, by manually introducing fewer fonts.  

The reasoning for using symlinks is to keep file system clean, so I can keep my [Google Fonts git repo](https://github.com/google/fonts) in a directory hierarchy _with other code_ -- where I can keep it manually updated near other git repos.

**Comment by JerryCasiano** [comment]
Well, the entire point of allowing users to specify directories in the sources tab of the preference pane is intended to facilitate this, but you are of course welcome to manage your files anyway you please.

As far as adding too many fonts goes I usually test with the entire Google fonts catalog plus an additional five thousand fonts enabled through the sources pane and while that amount of fonts certainly causes some slowdown it's nowhere near a freeze on my decade old laptop.

Anyway, closing this out as it's no longer an issue for the original reporter and I can't reproduce what you're describing.

**Comment by mdrmike** [comment]
Makes sense, appreciate you even looking into! People like you that maintain open source projects are gods & make the world go round! 

I found this issue based on a web search, and thought I'd share how I resolved for the benefit of community, or for my future self. 😊

That said, I'm a little surprised its not reproduce-able? Possibly because I used APT to install font-manager (version 8.8), instead of using Ubuntu Software Center. I do this for majority of programs I trust developers, but not all. I dont know, but here's a little more about my setup in case it provides a clue to help someone in the future.

Ubuntu 22.04.1. In general, I don't feel I have a lot of apps installed beyond vanilla Ubuntu, but I do have docker, VirtualBox which are probably the biggest changes. I also have MS Visual Studio Code, Darktable, Inkscape, Gimp. I'm also Dual booting with Windows.  


```
Ubuntu 22.04.1
Linux 5.15.0-56-generic x86_64 x86_64 GNU/Linux

Dell Inspiron 16 7620  (bought 3-4 months ago)
16GB RAM
2TB (upgraded:  WD_BLACK 2TB SN770 NVMe SSD - Gen4 PCIe, M.2 2280)* possibly the culprit? idk, but system has otherwise been solid. 
```
